### ISSUE TITLE
Tag Doesn't Work in 1.20.1 And Later Versions. 

### ISSUE THREAD

**Issue by KnightMithridatis** [McJty/LostCities#667]
For some reason, I cannot assign tags to blocks in version 1.20.1. So I tried all kinds of commands. For example, I set a tag to put 10 coals in the furnace. Unfortunately it didn't spawn.
![2024-12-29 154915](https://github.com/user-attachments/assets/cb89935f-054b-4c0d-8345-95b2e37d5205)
![2024-12-29 154951](https://github.com/user-attachments/assets/97bea297-5683-42a1-8db7-4abf12977028)
![2024-12-29 155037](https://github.com/user-attachments/assets/4b6ddfcf-cd32-4701-939c-fc5ae0ffa18c)

**Comment by KnightMithridatis** [comment]
The problem is definitely related to the Lost Cities mod. Maybe the Lost Cities tag code may be incompatible with version +1.20.

**Comment by KnightMithridatis** [comment]
Because in version 1.18.2, I can assign tags to blocks without any problems.

**Comment by McJty** [comment]
You cannot assign tags to blocks within an array like you're trying to do. i.e. if you have a palette entry with multiple possible blocks then those blocks cannot have tags. That's simply not supported

**Comment by KnightMithridatis** [comment]
> You cannot assign tags to blocks within an array like you're trying to do. i.e. if you have a palette entry with multiple possible blocks then those blocks cannot have tags. That's simply not supported

So, is there another way to add tags in 1.20.1 Lost Cities mod? Because it worked in 1.18.2. I can't even put coal in furnace anymore in 1.20.1. It's a bit absurd.

**Comment by McJty** [comment]
No, this didn't work in 1.18.2 either. This hasn't changed. Lost Cities has never been able to assign tags to blocks that are part of a single character block array. You can assign tags to blocks that are independent of course. Both in 1.18 and 1.20. But not to blocks in a randomizer

**Comment by KnightMithridatis** [comment]
> No, this didn't work in 1.18.2 either. This hasn't changed. Lost Cities has never been able to assign tags to blocks that are part of a single character block array. You can assign tags to blocks that are independent of course. Both in 1.18 and 1.20. But not to blocks in a randomizer

It works on Lost Cities version 1.18.2. I'm making highways with TrafficCraft. I need to adjust the color of the blocks with tags. As evidence I can offer this:
![2024-12-30 092523](https://github.com/user-attachments/assets/5938ae6f-5667-45ec-835e-4edde74a48a6)
![2024-12-30 092611](https://github.com/user-attachments/assets/52475673-bbe2-477b-823e-f274f0ee3f25)

**Comment by McJty** [comment]
You don't understand. That's different. There you are using the tag on a block that has no variations. That character is always assigned to that block. And then tags work. You cannot use tags on blocks that have variations. i..e multiple possible blocks for the same palette entry

**Comment by KnightMithridatis** [comment]
> You don't understand. That's different. There you are using the tag on a block that has no variations. That character is always assigned to that block. And then tags work. You cannot use tags on blocks that have variations. i..e multiple possible blocks for the same palette entry

So I understand you won't help.

**Comment by McJty** [comment]
What? I'm trying to help. I'm explaining what's going on. Tags work on standalone blocks. They don't work on randomized blocks. That was also like that in 1.18. That code literally hasn't changed

**Comment by McJty** [comment]
We have just confirmed that tags on a standalone block still work. Like the ";" that is defined in default.json from Lost Cities itself

**Comment by KnightMithridatis** [comment]
> We have just confirmed that tags on a standalone block still work. Like the ";" that is defined in default.json from Lost Cities itself

So can you give me a code example?

**Comment by KnightMithridatis** [comment]
> We have just confirmed that tags on a standalone block still work. Like the ";" that is defined in default.json from Lost Cities itself

> We have just confirmed that tags on a standalone block still work. Like the ";" that is defined in default.json from Lost Cities itself

{
      "char": ";",
      "block": "minecraft:furnace",
      "damaged": "minecraft:iron_bars",
      "tag": {
        "Items": [
          {
            "Slot": 0,
            "id": "minecraft:coal",
            "Count": 10
          }
        ]
      }
    },


I assigned this to the code in default.json, but it still didn't work.

**Comment by McJty** [comment]
That entry is by defaulit already present in default.json (from Lost Cities). Not sure why it isn't working for you though

**Comment by KnightMithridatis** [comment]
> That entry is by defaulit already present in default.json (from Lost Cities). Not sure why it isn't working for you though

I did it via kubejs. But kubejs works.

**Comment by McJty** [comment]
Can you show me your datapack where you are trying to do this?

**Comment by KnightMithridatis** [comment]
> Can you show me your datapack where you are trying to do this?

I'm trying another way. If it doesn't work, I'll write.

**Comment by KnightMithridatis** [comment]
> Can you show me your datapack where you are trying to do this?

I placed a separate generic block in default.json. It worked. Now I will try it for trafficcraft blocks.

![2024-12-30 105150](https://github.com/user-attachments/assets/0ad570ee-92e5-40ae-869b-a7a1e8d8a844)

![2024-12-30 105151](https://github.com/user-attachments/assets/905cc821-985a-41d9-864d-e98520c0b1d9)

**Comment by KnightMithridatis** [comment]
> Can you show me your datapack where you are trying to do this?

Unfortunately, whatever you did to the code in Lost Cities version 1.20-7.3.5. I can't tag the trafficcraft asphalt pattern block. However, I can assign tags in Lost Cities version 1.20-7.1.8. I share the pictures below, one by one.
![Ekran görüntüsü 2024-12-30 133643](https://github.com/user-attachments/assets/80bd59e7-bd18-4a22-bb72-a1873a76b48b)
![Ekran görüntüsü 2024-12-30 133413](https://github.com/user-attachments/assets/dceb18e0-10d5-4f4a-b178-bea91814f88b)
![Ekran görüntüsü 2024-12-30 133606](https://github.com/user-attachments/assets/339549d6-35aa-4ebe-ae3d-d049904f8de0)

**Comment by KnightMithridatis** [comment]
In short, the problem isn't related to TrafficCraft. There is a problem with Lost Cities version 7.3.5.

**Comment by McJty** [comment]
Well I can't reproduce the problem with the latest version. I would still like to see your datapack

**Comment by KnightMithridatis** [comment]
> Well I can't reproduce the problem with the latest version. I would still like to see your datapack

So how can I get the datapack? What exactly do you want? I think you want the kubejs\data\lostcities folder.

**Comment by McJty** [comment]
Well it's your datapack. So I assume you know how to get it? It's me that need to get it from you :-)

**Comment by KnightMithridatis** [comment]
> Well it's your datapack. So I assume you know how to get it? It's me that need to get it from you :-)

I'll send you the .jar files of the mods I used. I will also send the kubejs\data\lostcities folder. Because I've never had anything to do with datapack. You can test it yourself by creating a world. You can find trafficcraft blocks in buildings. Test it first with 7.1.8 and then with 7.3.5.


[Mods for 1.20.1 forge.zip](https://github.com/user-attachments/files/18274522/Mods.for.1.20.1.forge.zip)
[kubejs.zip](https://github.com/user-attachments/files/18274523/kubejs.zip)

**Comment by KnightMithridatis** [comment]
I don't know how the problem was solved. But it was fixed. I have no idea how it was fixed. It's really interesting.

**Comment by KnightMithridatis** [comment]
> Well it's your datapack. So I assume you know how to get it? It's me that need to get it from you :-)

It was fixed. The problem was related to Traffic Craft. I also tested it with version 7.3.5. There is no problem.
